backend/src/test/scala/com/softwaremill/adopttapir/starter/StarterServiceITTest.scala

```diff
@@ -3,49 +3,206 @@ package com.softwaremill.adopttapir.starter
 import better.files.{FileExtensions, File => BFile}
 import cats.effect.IO
 import cats.effect.unsafe.implicits.global
+import cats.instances.list._
+import cats.syntax.parallel._
+import com.softwaremill.adopttapir.logging.FLogging
 import com.softwaremill.adopttapir.starter.api._
-import com.softwaremill.adopttapir.template.ProjectTemplateInTests
+import com.softwaremill.adopttapir.template.ProjectTemplate
 import com.softwaremill.adopttapir.test.BaseTest
-import org.scalatest.ParallelTestExecution
+import org.scalatest.{Assertions, BeforeAndAfterAll, ParallelTestExecution}
+import os.SubProcess
+import sttp.client3.{HttpURLConnectionBackend, Identity, SttpBackend, UriContext, asStringAlways, basicRequest}
 
-import java.io.File
+import java.io.{PrintWriter, StringWriter}
+import scala.collection.mutable
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration.{FiniteDuration, SECONDS}
+import scala.util.Using
+import scala.util.matching.Regex
 
-class StarterServiceITTest extends BaseTest with ParallelTestExecution {
-
-  for {
+object Setup {
+  val validConfigurations = for {
     effect <- EffectRequest.values
     server <- ServerImplementationRequest.values
     docs <- List(true, false)
     metrics <- List(true, false)
     json <- JsonImplementationRequest.values
     starterRequest = StarterRequest("myproject", "com.softwaremill", effect, server, addDocumentation = docs, addMetrics = metrics, json)
     starterDetails <- FormValidator.validate(starterRequest).toSeq
-  } {
-    it should s"return zip file containing working sbt folder with: $effect/$server/docs=$docs/metrics=$metrics/$json" in {
-      val service = createStarterService
-      sbtCompileTest(service.generateZipFile(starterDetails))
+  } yield starterDetails
+
+  implicit class StarterDetailsWithDescribe(details: StarterDetails) {
+    lazy val describe: String =
+      s"${details.serverEffect}/${details.serverImplementation}/docs=${details.addDocumentation}/metrics=${details.addMetrics}/${details.jsonImplementation}"
+  }
+
+  type TestFunction = Integer => Unit
```

Comment:
I think that:
`type TestFunction = Integer => Assertion` 
will be more descriptive. What do you think ?

Reply:
Done